spec/rails/sharding/core_spec.rb

```diff
@@ -40,9 +39,22 @@
   end
 
   describe '.configurations' do
+    before do
+      stub_const('ENV', ENV)
+    end
+
+    let(:shard_configurations) { YAML.load(ERB.new(File.read('spec/fixtures/shards.yml')).result) }
+
     it 'should load shards.yml scoped by the current Rails.env or explicitly passed environment' do
-      expect(described_class.configurations).to be == YAML.load_file('spec/fixtures/shards.yml')['development']
-      expect(described_class.configurations('test')).to be == YAML.load_file('spec/fixtures/shards.yml')['test']
+      expect(described_class.configurations).to be == shard_configurations['development']
+      expect(described_class.configurations('test')).to be == shard_configurations['test']
+    end
+
+    it 'should load shards.yml with environment variables in ERB tags' do
```

Comment:
Probably the pseudocode of the test you want to do here is:
1. Set/mock env variable with something unique/different
2. load configuration and check values of username/password